### PR TITLE
WIP: #127 merge duplicate metrics instead of adding them.

### PR DESCRIPTION
**Pull request by ioggstream**
## what does this PR do

When a new metric is added via store.Add:

  - checks if there's another metric with the same labels
  - if so, copies LabelValues in the new metric
  - pop the old one from the array

### CODE REVIEW

**Comment by ioggstream** [comment]
I signed it!

File: metrics/store.go [hunk 1]
```diff
@@ -26,13 +28,43 @@ func NewStore() (s *Store) {
...
+		// - discard old metric.
+		for i, v := range s.Metrics[m.Name] {
+			//
+			if v.Program != m.Program {
+				continue
+			}
+			if v.Type != m.Type {
+				continue
+			}
+			if reflect.DeepEqual(v.Keys, m.Keys) {
```
Comment (ioggstream) [inline metrics/store.go]: @jaqx0r I'm fixing this part up, but I'd appreciate your opinion on that. I spinned it on my test environment and it seems to do the game.

Comment (jaqx0r) [reply metrics/store.go]: This seems fine. :)  Thanks!

**Comment by jaqx0r** [comment]
I'd like a test in mtail/mtail_test.go that reproduces the original problem -- creates a program, loads it, checks the exported variables, then does that again, and checks that there are no duplicates in the output.

**Pull request merged**